test/membrane_rtc_engine/webrtc/track_sender_test.exs

```diff
@@ -0,0 +1,150 @@
+defmodule Membrane.RTC.Engine.WebRTC.TrackSenderTest do
+  use ExUnit.Case, async: true
+
+  import Membrane.ParentSpec
+  import Membrane.Testing.Assertions
+
+  require Membrane.Pad
+
+  alias Membrane.{Buffer, Pad}
+  alias Membrane.RTC.Engine.Endpoint.WebRTC.TrackSender
+  alias Membrane.RTC.Engine.Track
+  alias Membrane.Testing.{Pipeline, Sink, Source}
+
+  @track_id "track1"
+  @simulcast_encodings ["h", "m", "l"]
+  @stream_id "stream1"
+  @track_origin "generated"
+
+  describe "TrackSender" do
+    test "adds `is_keyframe` flag to each buffer" do
+      [
+        {Track.new(:video, @stream_id, @track_origin, :H264, 90_000, :raw, nil,
+           id: @track_id,
+           simulcast_encodings: @simulcast_encodings
+         ), false},
+        {Track.new(:video, @stream_id, @track_origin, :VP8, 90_000, :raw, nil,
+           id: @track_id,
+           simulcast_encodings: @simulcast_encodings
+         ), false},
+        {Track.new(:audio, @stream_id, @track_origin, :OPUS, 48_000, :raw, nil, id: @track_id),
+         true}
+      ]
+      |> Enum.each(fn {track, expected_is_keyframe_value} ->
+        test_is_keyframe(track, expected_is_keyframe_value)
+      end)
+    end
+
+    test "sends caps just once on given output pad" do
+      track =
+        Track.new(:video, @stream_id, @track_origin, :H264, 90_000, :raw, nil,
+          id: @track_id,
+          simulcast_encodings: @simulcast_encodings
+        )
+
+      {:ok, pipeline} =
+        build_video_pipeline(track, [])
+        |> Pipeline.start_link()
+
+      assert_sink_caps(pipeline, :sink_h, %Membrane.RTP{})
+      assert_sink_caps(pipeline, :sink_m, %Membrane.RTP{})
+      assert_sink_caps(pipeline, :sink_l, %Membrane.RTP{})
+
+      refute_sink_caps(pipeline, :sink_h, %Membrane.RTP{}, 0)
+      refute_sink_caps(pipeline, :sink_m, %Membrane.RTP{}, 0)
+      refute_sink_caps(pipeline, :sink_l, %Membrane.RTP{}, 0)
```

Comment:
Why do we need this `refute_sink_caps`?

Reply:
We have to make sure that we received caps just once i.e. our mailbox doesn't have more caps messages

Reply:
Shouldn't we also test if we send caps when we are in state `:playing`?